app/helpers/catalog_helper/textual_summary.rb

```diff
@@ -1,6 +1,7 @@
 module CatalogHelper::TextualSummary
   def tags_from_record
     tags = []
+    tag_categories = {}
```

Comment:
I think we can simplify the code a bit with:

```suggestion
    tag_categories = Hash.new { |h, k| h[k] = [] }
```

Then down below...